custom_components/ttlock/switch.py

```diff
@@ -0,0 +1,70 @@
+"""Switch setup for our Integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import lock_coordinators
+from .entity import BaseLockEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all the locks for the config entry."""
+
+    async_add_entities(
+        [
+            entity
+            for coordinator in lock_coordinators(hass, entry)
+            for entity in (
+                Autolock(coordinator),
+                Locksound(coordinator),
+            )
+        ]
+    )
+
+class Autolock(BaseLockEntity, SwitchEntity):
```

Comment:
bit nitpicky, but this feels like it should be in TitleCase
```suggestion
class AutoLock(BaseLockEntity, SwitchEntity):
```